packages/botonic-react/src/util/webchat.js

```diff
@@ -0,0 +1,151 @@
+import { INPUT } from '@botonic/core'
+
+import { Button } from '../components/button'
+import { WEBCHAT } from '../constants'
+import { getProperty, strToBool } from './objects'
+import { deepMapWithIndex } from './react'
+
+/**
+ * Returns the value of a property defined in bot's theme based on WEBCHAT.CUSTOM_PROPERTIES dictionary.
+ * It gives preference to nested defined properties (e.g.: header.style) over plain properties (e.g.: headerStyle).
+ * If property doesn't exist, returns the defaultValue.
+ */
+export const _getThemeProperty = theme => (
+  property,
+  defaultValue = undefined
+) => {
+  for (const [k, v] of Object.entries(WEBCHAT.CUSTOM_PROPERTIES)) {
+    if (v == property) {
+      const nestedProperty = getProperty(theme, v)
+      if (nestedProperty !== undefined) return nestedProperty
+      const plainProperty = getProperty(theme, k)
+      if (plainProperty !== undefined) return plainProperty
+      return defaultValue
+    }
+  }
+  return undefined
+}
+
+export class ButtonsDisabler {
```

Comment:
This is not a "util". A util is a veeeery generic function that can be used in different contexts. This function is really specific to our Buttons, it deserves it's own file next to Button component definition.

Reply:
moved to its own file inside components 👍